pkg/router/template/router.go

```diff
@@ -301,6 +302,21 @@ func genCertificateHostName(hostname string, wildcard bool) string {
 	return hostname
 }
 
+// Returns the list of endpoints for the given route's service
+// action argument further processes the list e.g. shuffle
+// The default action is in-order traversal of internal data structure that stores
+//   the endpoints (does not change the return order if the data structure did not mutate)
+func processEndpointsForAlias(alias ServiceAliasConfig, svc ServiceUnit, action string) []Endpoint {
+	endpoints := endpointsForAlias(alias, svc)
+	if strings.ToLower(action) == "shuffle" {
```

Comment:
It's a normal end var to be defined in the deployment config. No docs yet. It's likely an experiment right now. 

Both random and shuffle are probably fine. I felt shuffle leaves less chance of misinterpretation.